src/main/java/org/carlspring/cloud/storage/s3fs/S3FileSystemProvider.java

```diff
@@ -573,7 +661,31 @@ public void copy(Path source, Path target, CopyOption... options)
         String bucketNameTarget = s3Target.getFileStore().name();
         String keyTarget = s3Target.getKey();
 
-        s3Source.getFileSystem().getClient().copyObject(bucketNameOrigin, keySource, bucketNameTarget, keyTarget);
+        final String encodedUrl = encodeUrl(bucketNameOrigin, keySource);
+
+        final CopyObjectRequest request = CopyObjectRequest.builder()
```

Comment:
If the object to be copied is > 5GB you will have to perform the copy as a Multipart Copy

Reply:
To be implemented in #95.